### ISSUE TITLE
Performance on adaptive adjustment mechanism

### ISSUE THREAD

**Issue by GNN-zl** [mengliu1998/DeeperGNN#7]
Hi,
I want to test the performance on adaptive adjustment mechanism. If the code of adaptive adjustment mechanism is
```
pps = torch.stack(preds, dim=1)   # nxkxc
retain_score = self.proj(pps)     # nxkx1
retain_score = retain_score.squeeze()  # nxk
retain_score = torch.sigmoid(retain_score)
retain_score = retain_score.unsqueeze(1) # nx1xk
out = torch.matmul(retain_score, pps).squeeze() # nx1xc -> nxc
```
After i comment out these lines and use preds[-1] as out, the performance of test accuracy declines to 0.397. It is even lower than MLP. So, Are these  the code of adaptive adjustment mechanism?

**Comment by mengliu1998** [comment]
Hi,

Yes, this is the implementation of the adaptive adjustment mechanism.

If you use preds[-1] as out, the performance would degrade. However, it would still keep a good accuracy around 80 on Cora, as shown in Section 3.2 in our paper. I think the hyperparameters (such as lr, weight_decay) need to be tuned if you remove the adaptive adjustment mechanism.

Hope this helps. Thank you.

**Comment by GNN-zl** [comment]
Thanks for your reply.
The accuracy turns to 82.1 on Cora after I try to tune the weight_decay to 0.001.
